### ISSUE TITLE
realm server issue  SyncUser.login

### ISSUE THREAD

**Issue by RawanA** [realm/realm-mobile-platform#180]
i want to log in into realm file using this code in swift 
but this function does not seem to be accessed, but i can open realm dashboard using the same username and password in the browser .

any help plz ! 

```
SyncUser.logIn(with: .usernamePassword(username: username, password: password, register: false), server: URL(string: "http://myIP:9080/")!) { user, error in
            guard let user = user else {
                fatalError(String(describing: error))
            }
            DispatchQueue.main.async {
                let configuration = Realm.Configuration(
                    syncConfiguration: SyncConfiguration(user: user, realmURL: URL(string: "realm://myIP:9080")!)
                )
                self.realm = try! Realm(configuration: configuration)
            }
        }
```

## Version of Realm and Tooling

- Realm Object Server Version: ?
- Flavor:
  - [ ] Developer
  - [ ] Professional
  - [ ] Enterprise
- Server OS & Version: ? (e.g. CentOS 6)
- Client SDK Version: ?
- Client OS & Version: ?

## Logs

<details>
  <summary>systemctl status realm-object-server.service</summary>
  <code>
    Paste output here
  </code>
</details>

<details>
  <summary>sudo journalctl -u realm-object-server.service</summary>
  <code>
    Paste output here
  </code>
</details>

<details>
  <summary>cat /var/log/realm-object-server.log</summary>
  <code>
    Paste output here
  </code>
</details>

**Comment by RiverbayChris** [comment]
Your realmURL in the sync config doesn't point to a specific realm, just the server.  You need to open via a specific path like /~/someRealmName.

**Comment by RawanA** [comment]
i changed the path into realm://myIP:9080/userID 
still having the same problem 
in the class there is 
public var realm : Realm!
but once i change that into 
var realm = try! Realm() 
SyncUser.login is accessed but nothing is added in the database in the server 
just added to the default realm file in the application directory

**Comment by radu-tutueanu** [comment]
@RawanA Try to change the url to  `realm://myIP:9080/~/realmFileName`. For example,  `realm://127.0.0.1:9080/~/firstRealm`
This creates a realm in the namespace of the current user.`~` will be automatically expanded to the user's id on the server. You can find this detailed in https://realm.io/docs/swift/latest/#opening-a-synchronized-realm .
Please let me know if this solves your issue or if anything is unclear.

**Comment by RawanA** [comment]
@radu-tutueanu and @RiverbayChris 
thank you guys for your response ^_^
the problem was solved finally in #181
